todos/todos.js

```diff
@@ -16,36 +16,52 @@ const logoutButton = document.querySelector('#logout');
 const deleteButton = document.querySelector('.delete-button');
 
 // let some todo state (an array)
+let todos = [];
 
 todoForm.addEventListener('submit', async (e) => {
-    // on submit, 
+    e.preventDefault();
+    const formData = new FormData(todoForm);
+    const todo = formData.get('todo');
+    // on submit,
     // create a todo in supabase using for data
+    await createTodo(todo);
+    todoForm.reset();
+
     // reset the form DOM element
     // and display the todos
+    fetchAndDisplayItems();
 });
 
-async function displayTodos() {
-    // clear the container (.textContent = '')
-    // fetch the user's todos from supabase
-
-    // loop through the user's todos
-        // for each todo, render a new todo DOM element using your render function
-        // then add an event listener to each todo
-            // on click, update the todo in supabase
-            // then (shockingly!) call displayTodos() to refresh the list
-        // append the rendered todo DOM element to the todosEl
-}
-
 window.addEventListener('load', async () => {
     // fetch the todos and store in state
     // call displayTodos
 });
 
+async function fetchAndDisplayItems() {
+    todos = await getTodos();
+    todosEl.textContent = '';
+    
+
+    for (let todo of todos) {
+        const todosAdded = renderTodo(todo);
+
+        todosAdded.addEventListener('click', async () => {
+            await completeTodo(todo.id);
+
+            fetchAndDisplayItems();
+        });
+
+        todosEl.append(todosAdded);
+    }
+}
+
 logoutButton.addEventListener('click', () => {
     logout();
 });
 
 deleteButton.addEventListener('click', async () => {
     // delete all todos
+    deleteAllTodos();
```

Comment:
This is an asynchronous function so we'll want to await here. Also - your deleteAllTodos() in fetch-utils takens in an  id so you'll want to pass in the idea here